src/client/scripts/game/chess/selection.js

```diff
@@ -124,6 +124,10 @@ const selection = (function() {
             return;
         }
 
+        // Don't allow players to move a piece of a color not matching the current player's turn
+        const selectedPieceColor = math.getPieceColorFromType(pieceSelected.type);
+        if (selectedPieceColor !== gamefile.whosTurn) return;
```

Comment:
Hey! Are these few lines needed! I can't find a reason, as `hoverSquareLegal` is always false if we've selected a friendly piece when it's not our turn.

Reply:
You are right that they are not really needed right now, because I added `selectionColorAgreesWithMoveTurn` to line 253, setting `hoverSquareLegal` to false in that case. My thinking was that we will want to remove that check in line 253 in the future again when we implement premoves because we will want to show the ghost pieces for the premoves, so `hoverSquareLegal` will need to be true, even when it is not our turn. In that case, we will need the currently superfluous lines again to prevent us from actually submitting a real move, even when a ghost piece is shown - at least that was my thinking. (On the other hand, maybe we will want to allow the move and will just handle it differently in the game logic, so nevermind all that and feel free to delete those supetfluous lines)

Reply:
I admit this is a bit half-baked, because I tried to anticipate the next steps for the premove implementation and forgot about these lines being superfluous for now.

Reply:
Alright! I added a new line, slightly modified, to accomplish this purpose!